incl/lib/mainLib.php

```diff
@@ -662,11 +662,11 @@ public function songReupload($url){
 		if (filter_var($song, FILTER_VALIDATE_URL) == TRUE && substr($song, 0, 4) == "http") {
 			$song = str_replace(["?dl=0","?dl=1"],"",$song);
 			$song = trim($song);
-			$query = $db->prepare("SELECT count(*) FROM songs WHERE download = :download");
+			$query = $db->prepare("SELECT id FROM songs WHERE download = :download");
 			$query->execute([':download' => $song]);	
-			$count = $query->fetchColumn();
-			if($count != 0){
-				return "-3";
+			$id = $query->fetchColumn();
+			if($id != false){
+				return $id;
```

Comment:
Your code lacks the song existence check.

Reply:
I am not a good PHPer, but `$id != false` should do the trick

Reply:
I see no problems with your code but since it returns an ID, it makes the detection of the already existence of the song.

Reply:
I see no issues with your code.